Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -57,15 +57,32 @@ bool LogFileReader::readAll(const QString& fileName)
  * that the conversion from string to double is sucessful.*/
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
+    bool conversionFailed = true;
     QStringList sections = line.split(BATDATA_DELIMITER);
 
-    QString timeString = sections.at(0);
-    batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
-
-    batteryData.voltage = sections.at(1).toDouble();
+    if (sections.size() == 3)
+    {
+        QString timeString = sections.at(0);
+        batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
+        if (!batteryData.time.isValid())
+        {
+            return false;
+        }
 
-    batteryData.current = sections.at(2).toDouble();
+        batteryData.voltage = sections.at(1).toDouble(&conversionFailed);
+        if (conversionFailed == false)
+        {
+            return false;
+        }
 
-    return true;
+        batteryData.current = sections.at(2).toDouble(&conversionFailed);
+        if(conversionFailed == false)
+        {
+            return false;
```

Comment:
There are a lot of return statements here, which is sort of a code smell. Try to combine all of these return falses into one if statement.